lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -415,28 +422,50 @@ createTransaction w (ApiT wid) body = do
     selection <- liftHandler $ W.createUnsignedTx w wid outs
     (tx, meta, wit) <- liftHandler $ W.signTx w wid pwd selection
     liftHandler $ W.submitTx w wid (tx, meta, wit)
-    return ApiTransaction
-        { id = ApiT (txId @t tx)
-        , amount = meta ^. #amount
-        , insertedAt = Nothing
-        , depth = Quantity 0
-        , direction = ApiT (meta ^. #direction)
-        , inputs = NE.fromList (coerceTxOut . snd <$> selection ^. #inputs)
-        , outputs = NE.fromList (coerceTxOut <$> W.outputs @t tx)
-        , status = ApiT (meta ^. #status)
-        }
+    return $ mkApiTransaction (txId @t tx)
+        (fmap Just <$> selection ^. #inputs) (selection ^. #outputs) meta
+
+mkApiTransaction
+    :: forall t.
+       Hash "Tx"
+    -> [(TxIn, Maybe TxOut)]
+    -> [TxOut]
+    -> W.TxMeta
+    -> ApiTransaction t
+mkApiTransaction txid ins outs meta = ApiTransaction
+    { id = ApiT txid
+    , amount = meta ^. #amount
+    , insertedAt = Nothing
```

Comment:
Why not take `insertedAt` and `depth` as parameters directly instead of updating them later?

Reply:
It looks like the answer is that `createTransaction` wants Nothing/0 as values.

Potential solution:
https://github.com/input-output-hk/cardano-wallet/pull/new/anviking/465/small-suggestions

but I actually fear that would be even worse as the `(Quantity 0) Nothing` parameters without labels is obfuscating.

Reply:
Yeah - I could go either way with those function params.